packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts

```diff
@@ -178,7 +178,7 @@ export class UsagePlan extends Resource {
    * @param apiKey
    */
   public addApiKey(apiKey: IApiKey): void {
-    new CfnUsagePlanKey(this, 'UsagePlanKeyResource', {
+    new CfnUsagePlanKey(this, `UsagePlanKey${apiKey.node.id}`, {
```

Comment:
```suggestion
    const prefix = `UsagePlanKey`;
    const id = this.node.tryGetChild(prefix) ? `${prefix}:${apiKey.node.uniqueId}` : prefix;
    new CfnUsagePlan(this, id, {
```

Reply:
Thanks. Will get this in.

Why do you suggest `uniqueId` over `id`? 
A scope-wide uniqueness is sufficient here, while, using `uniqueId` would make the physical id of this resource really long.